recipe/build.sh

```diff
@@ -68,6 +68,62 @@ if [[ "$blas_impl" == "accelerate" ]]; then
         -Wl,-reexport_library,$SRC_DIR/accelerate/liblapacke-netlib.${PKG_VERSION}.dylib
 
     cp libvecLibFort-ng.dylib $SRC_DIR/accelerate/
+
+elif [[ "$blas_impl" == "newaccelerate" ]]; then
+    # New Accelerate libraries have all symbols in BLAS, CBLAS, LAPACK with $NEWLAPACK
+    # name appended to all the symbol names. Therefore we create a library that dispatches
+    # eg: _dgemm -> _dgemm$NEWLAPACK with aliases.txt
+    #
+    # One exception to this is {c,z}dot{u,c}_ symbols which have different signatures.
+    # We use wrap_accelerate.c to fix those.
+    #
+    # For LAPACKE symbols, we use the LAPACKE wrappers from netlib which would call
+    # the LAPACK symbols from Accelerate.
+    #
+    # All of tese are exported from libblas_reexport.dylib
+
+    mkdir -p $SRC_DIR/accelerate
+    cp $NEW_ENV/lib/liblapacke.dylib $SRC_DIR/accelerate/liblapacke-netlib.${PKG_VERSION}.dylib
+    $INSTALL_NAME_TOOL -id "@rpath/liblapacke-netlib.${PKG_VERSION}.dylib" $SRC_DIR/accelerate/liblapacke-netlib.${PKG_VERSION}.dylib
+
+    veclib_loc=$SDKROOT/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A
+    veclib_libblas="${veclib_loc}/libBLAS.tbd"
+    veclib_liblapack="${veclib_loc}/libLAPACK.tbd"
+    if [[ ! -f ${veclib_loc}/libLAPACK.tbd ]]; then
+        echo "could not find TBD file ${veclib_loc}/libLAPACK.tbd"
+	exit 1
+    fi
+
+    export LDFLAGS="${LDFLAGS/-Wl,-dead_strip_dylibs/}"
+
+    for f in $veclib_libblas $veclib_liblapack; do
+      symbols=$(cat $f | grep -o '[a-z0-9_]*$NEWLAPACK' | rev | cut -b 11- | rev | sort | uniq)
+      for symbol in $symbols; do
+        if [[ "$symbol" != "_appleblas"* && "$symbol" != "_catlas"* && "$symbol" != "roc" && "$symbol" != "_cdot"* && "$symbol" != "_zdot"* ]]; then
+          echo $symbol'$NEWLAPACK' ${symbol} >> aliases.txt
+          if [[ "$symbol" != "cblas"* ]]; then
+            # Add _dgemm_ alias in addition to _dgemm
+            echo $symbol'$NEWLAPACK' ${symbol}_ >> aliases.txt
+          fi
+        fi
+      done
+    done
+    # These are defined in wrap_accelerate.c. Add a alias with the trailing underscore
+    echo _cdotu _cdotu_ >> aliases.txt
+    echo _cdotc _cdotc_ >> aliases.txt
+    echo _zdotu _zdotu_ >> aliases.txt
+    echo _zdotc _zdotc_ >> aliases.txt
```

Comment:
> I don't think the underscore-arithmetic is obvious here; wrap_accelerate.c specifies [cz]dot[cu] without any underscore, and as a C program, will get an unmangled symbol.

Not true. All C and Fortran names are prepended with an underscore and Fortran symbols are appended an underscore as well.

Reply:
C does not add an underscore, at least within the way the actual binary calls external symbols ([godbolt](https://godbolt.org/z/hzG61zors)). I clearly tried to research when underscores get added in Fortran (i.e., not always...). It's possible that there are other reasons why the leading underscore is added (e.g. by/for the linker?), but those are not obvious to me. 

I'm trying to get more of our foundational infrastructure documented in a way that makes it more accessible to other core members and contributors, so could you please be a bit more constructive than "not true"?

Reply:
> I'm trying to get more of our foundational infrastructure documented in a way that makes it more accessible to other core members and contributors, so could you please be a bit more constructive than "not true"?

You seem to believe that I owe you (or the contributors) an explanation. I don't. So, please be less aggressive.

cc @conda-forge/core

Reply:
Running your example on macos
```
% nm -g asd.o
0000000000000044 T _foo
                 U _func_int
                 U _func_long
                 U _func_long_long
0000000000000000 T _main
```

Reply:
> You seem to believe that I owe you (or the contributors) an explanation.

That's an interesting way to frame the regular review process. 🤔

> So, please be less aggressive.

I'm perplexed by this insinuation - can you point out what you perceived as aggressive in my comment? In any case, I suggest to take this to the core chat.

Reply:
> Running your example on macos

Can we add something like "due to the default calling convention on osx"? Below's a suggestion:
```suggestion
    # The functions cdotu etc. are defined in wrap_accelerate.c; due to the default
    # calling convention on osx, the respective symbols get prefixed with an underscore.
    # Fortran uses yet another default that generally requires a trailing underscore;
    # add an alias for that so that the symbols are found from both languages.
    echo _cdotu _cdotu_ >> aliases.txt
    echo _cdotc _cdotc_ >> aliases.txt
    echo _zdotu _zdotu_ >> aliases.txt
    echo _zdotc _zdotc_ >> aliases.txt
```

PS. It's hard to find good sources for the leading "_", aside from some random & ancient SO [questions](https://stackoverflow.com/questions/2627511/why-do-c-compilers-prepend-underscores-to-external-names), and a bit of trivia in [wikipedia](https://en.wikipedia.org/wiki/Name_mangling#C).

Reply:
I don't think anyone was trying to escalate anything here. Good documentation is always welcome, but sometimes is costly.

This sentence here is what we should strive for, under ideal conditinos:

> [...] trying to get more of our foundational infrastructure documented in a way that makes it more accessible to other core members and contributors [...]

It's true that some of that knowledge might be difficult to source because it could be based on individual research, accumulated know-how through the years, etc. Documenting that down is no easy feat, and it's ok to say that we don't have the resources available to go down that path. Sometimes a hint on what to search for is enough too.

For now let's see if the suggestions are accurate and provide value for the current and future maintainers.